src/mscorlib/shared/System/Number.Formatting.cs

```diff
@@ -1130,8 +1130,7 @@ private static unsafe bool TryCopyTo(char* src, int length, Span<char> destinati
         {
             if (length <= destination.Length)
```

Comment:
Same suggestion here: optimistically call TryCopyTo and branch on its return statement.